test-utils/client/src/lib.rs

```diff
@@ -155,16 +159,14 @@ impl<Block: BlockT, Executor, Backend, G: GenesisInit> TestClientBuilder<Block,
 	}
 
 	/// Set the execution strategy that should be used by all contexts.
-	pub fn set_execution_strategy(
-		mut self,
-		execution_strategy: ExecutionStrategy
-	) -> Self {
-		self.execution_strategies = ExecutionStrategies {
-			syncing: execution_strategy,
-			importing: execution_strategy,
-			block_construction: execution_strategy,
-			offchain_worker: execution_strategy,
-			other: execution_strategy,
+	pub fn set_execution_strategy(mut self, execution_strategy: ExecutionStrategy) -> Self {
+		self.execution_configs = ExecutionConfigs {
+			// NOTE: we set the context to consensus for all of them.
+			syncing: execution_strategy.in_consensus(),
+			importing: execution_strategy.in_consensus(),
+			block_construction: execution_strategy.in_consensus(),
+			offchain_worker: execution_strategy.in_consensus(),
```

Comment:
shouldn't offchain be 'execution_strategy.in_offchain'.
In fact I would find it ok to have all other using 'execution_strategy.into()' doing 'in_consensus' : so we only really care for specific offchain case.
But that is just because as a api user I would not really want to learn what is 'consensus', just that offchain is different.
But this is highly discutable, current api seems fine.

Reply:
Rereading my comments, I find 'in_consensus' ok.

Reply:
yeah I replicated the old logic here, but will set it to offchain now. Also note that this is all used only in testing anyhow.